src/internal/connector/onedrive/url_cache.go

```diff
@@ -0,0 +1,275 @@
+package onedrive
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/alcionai/clues"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+)
+
+type itemProperties struct {
+	downloadURL string
+	isDeleted   bool
+}
+
+// urlCache caches download URLs for drive items
+type urlCache struct {
+	driveID string
+	// urlMap stores Item ID -> item property map
+	urlMap          map[string]itemProperties
+	lastRefreshTime time.Time
+	refreshInterval time.Duration
+	// cacheLock protects urlMap and lastRefreshTime
+	cacheLock sync.RWMutex
+	// refreshMutex serializes cache refresh attempts
+	refreshMutex    sync.Mutex
```

Comment:
`mu` for standard golang naming expectations.
```suggestion
	cacheMu sync.RWMutex
	// refreshMutex serializes cache refresh attempts
	refreshMu    sync.Mutex
```